service/lxd.go

```diff
@@ -742,6 +785,14 @@ func (s *LXDService) waitReady(ctx context.Context, c lxd.InstanceServer, timeou
 
 			_, _, err := c.RawQuery("GET", "/internal/ready", nil, "")
 			if err != nil {
+				if api.StatusErrorCheck(err, http.StatusNotFound) {
```

Comment:
In case `c.RawQuery` returns a non nil error which does not match `http.StatusNotFound` it will loop forever.
This should not be a problem as the outer func exits after the provided parent ctx gets cancelled but this will leak the routine as it never exits. 

Maybe to go routine can be structured like this to ensure it exits after the context is cancelled:

```go
go func() {
    for i := 0; ; i++ {
        select {
            case <-unixCtx.Done():
                return // Exit the routine
            default:
                // run the actual logic
        }
    }
}()
```

Reply:
Not quite, `unixCtx` being done isn't sufficient to exit the goroutine, it's only a premature check in case we can't find the unix socket. 

Instead I think we can just check if `ctx.Err` is not nil and exit if so in each iteration.

Reply:
Additionally, I just noticed the logger is useless since this is run on the client side anyway.

Reply:
I've also set a default 1 minute timeout in the `waitReady` function itself in case a caller forgets to set one on the context. This is also mentioned in the description.